pkg/tsdb/prometheus/resource/resource.go

```diff
@@ -20,6 +22,46 @@ type Resource struct {
 	log      log.Logger
 }
 
+// Hop-by-hop headers. These are removed when sent to the backend.
+// http://www.w3.org/Protocols/rfc2616/rfc2616-sec13.html
+var hopHeaders = []string{
+	"Connection",
+	"Keep-Alive",
+	"Proxy-Authenticate",
+	"Proxy-Authorization",
+	"Te", // canonicalized version of "TE"
+	"Trailers",
+	"Transfer-Encoding",
+	"Upgrade",
+}
+
+// These headers simply do not work when set from the client, so
+// strip them out before sending.
+var removeHeaders = []string{
+	"accept",
+	"Accept-Encoding",
```

Comment:
I think this makes sense but I wonder if we should send the accept gzip header ourselves here? Seems like that would be actually reasonable.